backend/open_webui/grounding/wiki_search_utils.py

```
@@ -0,0 +1,689 @@
+            }
+
+        available = (
+            cls._semaphore._value if hasattr(cls._semaphore, "_value") else "unknown"
+        )
+        waiters = getattr(cls._semaphore, "_waiters", None)
+        waiting = len(waiters) if waiters is not None else "unknown"
+
+        return {
+            "semaphore_initialized": True,
+            "available_permits": available,
+            "max_concurrent": max_concurrent,
```

Ruff (0.12.2)

Undefined name `max_concurrent`

(F821)